app/src/main/java/org/piramalswasthya/cho/ui/commons/case_record/CaseRecordCustom.kt

```
@@ -314,62 +357,137 @@ class CaseRecordCustom : Fragment(R.layout.case_record_custom_layout), Navigatio
             getVisitResObserver(benVisitInfo)
+
+            // Use latest pharmacist_flag from DB so completed (lab+pharmacist) cases get correct UI even if intent was stale
+            val benVisitNo = benVisitInfo.benVisitNo
+            val sync = if (benVisitNo != null) {
+                runBlocking(Dispatchers.IO) {
+                    try {
+                        viewModel.getPatientVisitInfoSyncByPatientIdAndBenVisitNo(
+                            benVisitInfo.patient.patientID,
+                            benVisitNo
+                        )
+                    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)